### PR TITLE
feat: add aten.glu converter

### LINKED ISSUE #2663: ↔ [Converter] Add support for `torch.ops.aten.glu.default` in Torch-TensorRT
Add support for `torch.ops.aten.glu.default`, documented [here](https://pytorch.org/docs/stable/generated/torch.nn.functional.glu.html) and [here](https://github.com/pytorch/pytorch/blob/f3dd2a544c787aa11bcb84837a50195864f73cc9/aten/src/ATen/native/native_functions.yaml#L11564).

See #2197 for usecase.

### PR DESCRIPTION
**Pull request by zupengwang**
# Description

Adds Dynamo converter support for `torch.ops.aten.glu.default`. The converter implements GLU as an equal split followed by sigmoid and elementwise multiplication, and conservatively requires the split dimension to be statically known and even while allowing other dimensions to remain dynamic.

This avoids a PyTorch fallback and extra TensorRT engine partitions for models that use GLU, including the Conformer use case described in #2197. No additional dependencies are required.

Validation performed locally:

- Converter tests cover FP32, FP16, `dim=0/1/-1`, rank 2/3, and a dynamic batch range.
- GLU, split, and sigmoid converter tests: 17 passed.
- Full-compilation `torch.export` checks on two RTX 3090 GPUs produced a single TensorRT module with no PyTorch fallback for FP32, FP16, and dynamic shapes; outputs matched eager PyTorch exactly in the tested cases.
- Black, Ruff, isort, mypy, and `git diff --check` passed for the changed files.

The full repository test suite and GitHub CI have not been run. The dynamic GLU split dimension remains unsupported; invalid odd split dimensions retain PyTorch's existing error behavior.

Fixes #2663

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks (no non-obvious hacks were added)
- [x] I have made corresponding changes to the documentation (not required; supported-op documentation is generated from the converter registry)
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py [hunk 1]
```diff
@@ -488,6 +488,54 @@ def aten_ops_hardtanh(
...
+            "split dimension, falling back to PyTorch operation."
+        )
+        return False
+
+    input_shape = input_val.shape
+    dim = get_positive_dim(args_bounds_check(node.args, 1, -1), len(input_shape))
+    split_dim_size = input_shape[dim]
+
+    return isinstance(split_dim_size, int) and split_dim_size % 2 == 0
+
```
Comment (apbose) [inline py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:527]: dim passed as a keyword is silently ignored. Both glu_validator and aten_ops_glu read only node.args[1] / args[1] via args_bounds_check. If the graph carries dim as a kwarg instead  torch.fx.symbolic_trace on torch.ops.aten.glu.default(x, dim=0) produces args=(x,), kwargs={'dim': 0} both validator and converter silently fall back to dim=-1, producing a wrong split axis instead of erroring

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py [hunk 2]
```diff
@@ -488,6 +488,54 @@ def aten_ops_hardtanh(
...
+
```
Comment (apbose) [inline py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:528]: Zero-length split dimension passes validation, then crashes. glu_validator only checks split_dim_size % 2 == 0; 0 % 2 == 0 is True, so a zero-sized split axis passes. The converter then computes split_size = 0, and impl.split.split's num_splits = (dim_size + split_size - 1) // split_size divides by zero, crashing engine build

File: tests/py/dynamo/conversion/test_glu_aten.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        input_specs = [
+            Input(
+                min_shape=(2, 4, 8),
+                opt_shape=(3, 4, 8),
+                max_shape=(5, 4, 8),
+                dtype=torch.float32,
+            ),
+        ]
+        self.run_test_with_dynamic_shape(Glu(), input_specs, use_dynamo_tracer=True)
+
```
Comment (apbose) [inline tests/py/dynamo/conversion/test_glu_aten.py:67]: please add test cases for above, like dim passed as keyword, omitted dim, zero sized split axis, test exercising validator reject path

**Comment by lanluo-nvidia** [comment]
@zupengwang  
could you please take a look on the review comments if you want this PR to go in the 2.14 release.

**Comment by zupengwang** [comment]
Thanks for the reminder. Update: the review comments have been addressed and pushed in f0dae4231. GPU validation is now complete: the GLU converter suite passes 12 tests, and the neighboring split/sigmoid regression suites pass 13 tests on an RTX 3090.

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py (hunk 1 above)
Comment (zupengwang) [reply py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:527]: Addressed in f0dae4231. Both glu_validator and aten_ops_glu now use the shared _get_glu_dim helper, which handles positional, keyword, and omitted/default dim consistently.

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py (hunk 2 above)
Comment (zupengwang) [reply py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:528]: Addressed in f0dae4231. The validator now requires split_dim_size > 0 in addition to being a statically known even integer, so a zero-sized split axis is rejected before impl.split.split.

File: tests/py/dynamo/conversion/test_glu_aten.py (hunk 3 above)
Comment (zupengwang) [reply tests/py/dynamo/conversion/test_glu_aten.py:67]: Added focused coverage in f0dae4231 for keyword dim, omitted/default dim, zero-sized and odd split dimensions, missing metadata, and the validator rejection path. The GPU-backed GLU suite passes 12 tests, and neighboring split/sigmoid tests pass 13 tests on an RTX 3090.

**Comment by apbose** [comment]
LGTM

**Review by apbose — APPROVED** [review]
